### ISSUE TITLE
function with cosmos db output binding complains about extension not being installed

### ISSUE THREAD

**Issue by askpatrickw** [microsoft/vscode-azurefunctions#1403]
I added a second output binding to my function using the vscode extension.

When I try to run it locally, I get an error:
"The binding type(s) 'cosmosDB' are not registered. Please ensure the type is correct and the binding extension is installed."

I figured out that host.json was not updated ( related #1343 ) so I added:

```json
{
  "version": "2.0",
  "extensions": {
    "cosmosDB": {
        "connectionMode": "Gateway",
        "protocol": "Https"
    }
}
}
```

But, I still get the same error as before, no change.
I cannot find anything about installing a cosmos db binding extension.

## Environment
> Version: 1.36.1 (user setup)
> Commit: 2213894ea0415ee8c85c5eea0d0ff81ecc191529
> Date: 2019-07-08T22:59:35.033Z
> Electron: 4.2.5
> Chrome: 69.0.3497.128
> Node.js: 10.11.0
> V8: 6.9.427.31-electron.0
> OS: Windows_NT x64 10.0.18362

> (.env) PS: github\project> python --version
> Python 3.6.8

**Comment by EricJizbaMSFT** [comment]
@askpatrickw the `extensions` property has a different purpose. You'll want to add the `extensionBundle` property to your host.json. It contains several extensions, including Cosmos DB - see [here](https://docs.microsoft.com/en-us/azure/azure-functions/functions-bindings-register#extension-bundles) for more info.

```
{
  "version": "2.0",
  "extensionBundle": {
    "id": "Microsoft.Azure.Functions.ExtensionBundle",
    "version": "[1.*, 2.0.0)"
  }
}
```

NOTE: Make sure your func cli is up to date. This "extensionBundle" feature was just added last May - not sure of the specific version but I can check if you need me to

Assuming the above works, I think we can close this as a duplicate of #1343

**Comment by EricJizbaMSFT** [comment]
(added a comment with the workaround to #1343)

**Comment by askpatrickw** [comment]
func --version 2.7.1480 
This is latest according to https://github.com/Azure/azure-functions-core-tools/releases

**Comment by askpatrickw** [comment]
And now I get this error.

"Value cannot be null.
Parameter name: provider"

I deleted my function.
Created a new function bound to http. It starts fine.
I added an additional outbound binding of cosmosdb.
Debug will not start with the error I pasted above.

**Comment by EricJizbaMSFT** [comment]
Okay yeah I could reproduce that error. Try running this command from the VS Code Command palette:
![Screen Shot 2019-07-25 at 5 58 39 PM](https://user-images.githubusercontent.com/11282622/61918352-e731ab00-af05-11e9-921d-22601af4711b.png)

The functions runtime needs a storage account for internal use in order to debug and that command will help you set it. If that fixes it, we should probably file an issue on the func cli to have a better error message than "Value cannot be null. Parameter name: provider"

**Comment by askpatrickw** [comment]
Yep.. they did it.  My $0.0002.

1. Yeah, that error message could be way more helpful.
2. The docs are kind of spread out all over the place. And its non-obvious all the steps which need to happen. Maybe listing all the components which may need to be installed on the plugin page more like a check list. "You will need to install: 1) The plugin, 2) the tools, 3) maybe one of the following emulators. And a section on using storage with emulators or directly online. 

Just a sugestion… 

Thank you for the help though!  You unblocked me and I successfully wrote data to cosmosdb!

**Comment by EricJizbaMSFT** [comment]
Filed an issue on the error message here: https://github.com/Azure/azure-functions-core-tools/issues/1460

I can't help much on the docs, but we do have a few issues to improve this in VS Code, closing as a duplicate of those. https://github.com/microsoft/vscode-azurefunctions/issues/1343 https://github.com/microsoft/vscode-azurefunctions/issues/1332
